core/src/session.cpp

```diff
@@ -96,7 +96,7 @@ namespace irods::http
 		}
 
 		if (ec) {
-			return irods::fail(ec, "read");
+			return logging::error("{}: read: {}", __func__, ec.message());
```

Comment:
Should we log and *then* return like on L94-L95?

Reply:
I'm leaning toward log-then-return just for the sake of consistency, but I also do not feel too strongly about it.

Reply:
Split across two lines.